### ISSUE TITLE
Exception when decoding Jackson-encoded `Base64` binary value in YAML

### ISSUE THREAD

**Issue by tlrx** [FasterXML/jackson-dataformats-text#90]
Using jackson-dataformats-text-2.9.6 binary values that exceed a given length cannot be decoded back and the following exception is thrown:

```
com.fasterxml.jackson.core.JsonParseException: Illegal character '\' (code 0x5c) in base64 content
```

This is related to #62 which forces the `Base64Variant` to be `MIME`. When the encoded value exceeds a given size, line seperators are inserted into the encoded value but it then fails to be decoded.

I created the following small test and added it to the `SimpleGenerationTest` in `yaml`:

```java
public void testCase() throws Exception {
        final int length = 50;
        final byte[] data = new byte[length];
        Arrays.fill(data, (byte) 1);

        final YAMLFactory yamlFactory = new YAMLFactory();

        final ByteArrayOutputStream os = new ByteArrayOutputStream();
        try (JsonGenerator generator = yamlFactory.createGenerator(os, JsonEncoding.UTF8)) {
            generator.writeStartObject();
            generator.writeBinaryField("data", data);
            generator.writeEndObject();
        }

        try (JsonParser parser = yamlFactory.createParser(os.toByteArray())) {
            assertEquals(JsonToken.START_OBJECT, parser.nextToken());
            assertEquals(JsonToken.FIELD_NAME, parser.nextToken());
            assertEquals("data", parser.currentName());
            assertEquals(JsonToken.VALUE_EMBEDDED_OBJECT, parser.nextToken());
            assertArrayEquals(data, parser.getBinaryValue());
            assertEquals(JsonToken.END_OBJECT, parser.nextToken());
            assertNull(parser.nextToken());
        }
    }
```

This test passed correctly when `length` is small (`50`) but with larger size (like `100`) it fails.

**Comment by vboulaye** [comment]
I have the same problem encoding/decoding binary arrays longer than 76 characters after upgrading from 2.9.2 to 2.9.6.

I submitted a PR to try to fix it by changing the way the encoding is done in Base64Variant

**Comment by cowtowncoder** [comment]
@tlrx Thank you for reporting this; fixed for 2.9.9/2.10.0 now. And special thanks to @vboulaye for providing the fix here. Sorry it took this long to get simple thing fixed, but better late than never.
